Update persist init and generate commands with design v3 changes (#148)
ballerina-platform/persist-tools: docs/spec/spec.md

## 2. Initializing the Bal Project with Persistence Layer

```bash
bal persist init --datastore="datastore" --module="module_name"
```

 Following optional environment variables can be used with `bal persist init` command.
 - datastore : used to indicate the preferred database client. Currently, only 'mysql' is supported.

>  - module : used to indicate the preferred name for the schema file. By default, this will be equal to the package name.

Reviewer: Is this to indicate the schema file or to which module needs to be persistence enabled? 
@daneshk WDYT?

Reviewer 2: No this doesn't have any connection with the schema file. This indicates the module in which the persistence is enabled and relevant codes are generated.

Author: updated in [1ecce62](https://github.com/ballerina-platform/persist-tools/pull/148/commits/1ecce626e9420d36be876cbb13c560ef8a376d09)

## 2. Initializing the Bal Project with Persistence Layer

```bash
bal persist init --datastore="datastore" --module="module_name"
```

| Command Parameter |                                       Description                                        | Mandatory | Default Value |
|:-----------------:|:----------------------------------------------------------------------------------------:|:---------:|:-------------:|
|    --datastore    |  used to indicate the preferred database client. Currently, only 'mysql' is supported.   |    No     |     mysql     |

> |     --module      |      used to indicate the persist enabled module in which the files are generated.       |    No     |               |

Reviewer 2: ```suggestion
|     --module      |      used to indicate the persist enabled module in which the files are generated.       |    No     |     <package_name>          |
```

The command initializes the bal project with the persistence layer. This command includes the following steps,

1. Create persist directory
   This directory should contain all data model definition files. This file will define the required entities as per the [`persist` specification](https://github.com/ballerina-platform/module-ballerina-persist/blob/main/docs/spec/spec.md#2-data-model-definition)
2. Create a model definition file in persist directory

>    It will create a file with same name as the module environment variable if no files are present in the `persist` directory. It will create an empty file with required imports(`import ballerina/persist as _;`).

Reviewer 2: ```suggestion
   It will create a file named `model.bal` with required imports(`import ballerina/persist as _;`),  if no files are present in the `persist` directory.
```

3. Update Ballerina.toml with persist module configurations.
   It will update the Ballerina.toml file with persist configurations.
    ```ballerina
    [persist]

>     datastore = "datastore"

Reviewer 2: ```suggestion
    datastore = "mysql"
```

    module = "<package_name>.<module_name>"
   ```

> 4. Create(Update) Config.toml file inside the Ballerina project.

Reviewer 2: We don't update Config.toml file in init command.

Reviewer 2: Move this to generate command

Author: updated in [c6f8aa2](https://github.com/ballerina-platform/persist-tools/pull/148/commits/c6f8aa200f6cd6b2665e3f091a75d12c1ef59a0f)

|    --datastore    |  used to indicate the preferred database client. Currently, only 'mysql' is supported.   |    No     |     mysql     |
|     --module      |      used to indicate the persist enabled module in which the files are generated.       |    No     |     <package_name>          |

The command initializes the bal project with the persistence layer. This command includes the following steps,

1. Create persist directory
   This directory should contain all data model definition files. This file will define the required entities as per the [`persist` specification](https://github.com/ballerina-platform/module-ballerina-persist/blob/main/docs/spec/spec.md#2-data-model-definition)
2. Create a model definition file in persist directory
   It will create a file named `model.bal` with required imports(`import ballerina/persist as _;`),  if no files are present in the `persist` directory.
3. Update Ballerina.toml with persist module configurations.
   It will update the Ballerina.toml file with persist configurations.
    ```ballerina
    [persist]
    datastore = "mysql"
    module = "<package_name>.<module_name>"
   ```

The directory structure will be,
```
medical-center

> ├── generated

Reviewer 2: ```suggestion
```

## 2. Initializing the Bal Project with Persistence Layer

```bash

> bal persist init --datastore="datastore" --module="module_name"

Reviewer: Could you update the spec with what the datastore and module params are meant to do?

Author: updated in [723d216](https://github.com/ballerina-platform/persist-tools/pull/148/commits/723d216a68224db64d79e239a736b48849e4706f)

Reviewer: Shall we move it into table structure including following info?
1. Commnad
2. Description
3. Mandatory
4. Defaultvalue

Author: updated in [9da43ea](https://github.com/ballerina-platform/persist-tools/pull/148/commits/9da43ea9f1f785ffba65a086ba56ac2af1adefb3)

   This directory should contain all data model definition files. This file will define the required entities as per the [`persist` specification](https://github.com/ballerina-platform/module-ballerina-persist/blob/main/docs/spec/spec.md#2-data-model-definition)
2. Create a model definition file in persist directory
   It will create a file with same name as the package name if no files are present in the `persist` directory. It will create an empty file with required imports(`import ballerina/persist as _;`).

> 3. Create the generated directory. If the definition filename inside the `persist` directory is different from the Ballerina package name, create a submodule named as the definition file name.

Reviewer 2: generated directory and submodule should also create at generate command and the module name should be read from the Ballerina.toml file. In the init command, we only populate Ballerina.toml file with the provider and the module name.

Author: updated in [c5567e7](https://github.com/ballerina-platform/persist-tools/pull/148/commits/c5567e7273ef9c55ed1e47da998f94e65cf3a506)

> 4. Update Ballerina.toml with database configurations.

Reviewer 2: ```suggestion
4. Update Ballerina.toml with persist module configurations.
```

Author: updated in [c5567e7](https://github.com/ballerina-platform/persist-tools/pull/148/commits/c5567e7273ef9c55ed1e47da998f94e65cf3a506)

Behaviour of the `init` command,
- User should invoke the command within a bal project

> - User can use the optional environment variables to indicate the preferred module name and data store, otherwise default values will be used.

Reviewer: ```suggestion
- User can use the optional command options to indicate the preferred module name and data store, otherwise default values will be used.
```

> - User can use the optional environment variables to indicate the preferred module name and data store, otherwise default values will be used.

Reviewer 2: ```suggestion
- User can use the optional arguments to indicate the preferred module name and data store, otherwise default values will be used.
```

Author: updated in [c6f8aa2](https://github.com/ballerina-platform/persist-tools/pull/148/commits/c6f8aa200f6cd6b2665e3f091a75d12c1ef59a0f)

as per the `persist` specification.
Additionally, this command will create the database schema associated with the data model definition.

> It will add generated files under the conventions,

Reviewer: I dont think this part is valid anymore isnt it? We generate based on Ballerina.toml persist.module value dont we?

> 1. If the file name is the same as the package name, it will generate the files under the `default` module.

Reviewer 2: ```suggestion
1. If the model name is the same as the package name, it will generate the files under the `default` module.
```

Author: updated in [c6f8aa2](https://github.com/ballerina-platform/persist-tools/pull/148/commits/c6f8aa200f6cd6b2665e3f091a75d12c1ef59a0f)

## 4. Push Persistence Schema to the Data Provider

```bash

> bal persist push  #The is not supported.

Reviewer 2: ```suggestion
bal persist push  #The is not supported.
```
```suggestion
bal persist push
```

bal persist push
```

> 

Reviewer 2: Add a comment here saying `The is not supported.`

Author: updated in [c6f8aa2](https://github.com/ballerina-platform/persist-tools/pull/148/commits/c6f8aa200f6cd6b2665e3f091a75d12c1ef59a0f)
